### PR TITLE
Add example demo for ResNet with TensorFlow and DALI

### PR DESCRIPTION
**Pull request by Kh4L**
Signed-off-by: Serge Panev <spanev@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Maybe we can plug it into qa/L1_jupyter_plugins/test.sh or create separate test for that.

File: docs/examples/tensorflow/demo/nvutils/cmdline.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                   required=_required(init_vals, 'display_every'),
+                   help="""How often (in batches) to print out
+                   running information.""")
+    p.add_argument('--precision', choices=['fp32', 'fp16'],
+                   default=_default(init_vals, 'precision'),
+                   required=_required(init_vals, 'precision'),
+                   help="""Select single or half precision arithmetic.""")
+    p.add_argument('--dali_cpu', action='store_true',
+                   default=False,
+                   help="""Use CPU decoder for DALI for input pipeline""")
```
Comment (JanuszL) [inline docs/examples/tensorflow/demo/nvutils/cmdline.py]: I would say it is not only decoder but most of the pipeline is CPU based.

File: docs/examples/tensorflow/demo/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+concert with MPI. To train ResNet-50 (--layers=50) using 8 V100 GPUs, for example on DGX-1,
+use the following command:
+
+```
+$ mpiexec --allow-run-as-root --bind-to socket -np 8 python resnet.py \
+                                                     --layers=50 \
+                                                     --data_dir=/data/imagenet \
+                                                     --precision=fp16 \
+                                                     --log_dir=/output/resnet50 \
+                                                     --dali_cpu
```
Comment (JanuszL) [inline docs/examples/tensorflow/demo/README.md]: You could add info about dali_cpu option somewhere in the README as well.

Comment (Kh4L) [reply docs/examples/tensorflow/demo/README.md]: Done

File: docs/examples/tensorflow/demo/nvutils/cmdline.py (hunk 1 above)
Comment (Kh4L) [reply docs/examples/tensorflow/demo/nvutils/cmdline.py]: Done

File: docs/examples/tensorflow/demo/nvutils/runner.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+        np.random.seed(2)
+
+    log_dir  = None if log_dir  == "" else log_dir
+    data_dir = None if data_dir == "" else data_dir
+    if data_dir is None:
+        raise ValueError("data_dir must be specified")
+    if log_dir is None:
+        raise ValueError("log_dir must be specified")
+
+    filename_pattern = os.path.join(data_dir, '%s-*')
```
Comment (JanuszL) [inline docs/examples/tensorflow/demo/nvutils/runner.py]: It should be something:
```
    if data_dir is not None:
        filename_pattern = os.path.join(data_dir, '%s-*')
        eval_filenames = sorted(tf.gfile.Glob(filename_pattern % 'validation'))
        num_eval_samples = _get_num_records(eval_filenames)
    else:
        num_training_samples = global_batch_size
    eval_idx_filenames = None
    if data_idx_dir is not None:
        filename_pattern = os.path.join(data_idx_dir, '%s-*')
        eval_idx_filenames = sorted(tf.gfile.Glob(filename_pattern % 'eval'))
```

File: docs/examples/tensorflow/demo/nvutils/runner.py [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    except KeyboardInterrupt:
+        print("Keyboard interrupt")
+
+def validate(infer_func, params):
+    image_width = params['image_width']
+    image_height = params['image_height']
+    image_format = params['image_format']
+    batch_size = params['batch_size']
+    data_dir = params['data_dir']
+    log_dir = params['log_dir']
```
Comment (JanuszL) [inline docs/examples/tensorflow/demo/nvutils/runner.py]: Add:
```
data_idx_dir = params['data_idx_dir']
```

File: docs/examples/tensorflow/demo/nvutils/runner.py [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    if not deterministic:
+        num_preproc_threads = 2
+    else:
+        num_preproc_threads = 1
+
+    if hvd.rank() == 0:
+        print("Evaluating")
+        try:
+            eval_result = classifier.evaluate(
+                input_fn=lambda: nvutils.image_set(
```
Comment (JanuszL) [inline docs/examples/tensorflow/demo/nvutils/runner.py]:
```
    input_func = lambda: nvutils.image_set(
        eval_filenames, batch_size, image_height, image_width,
        training=False, distort_color=False,
        deterministic=deterministic, num_threads=num_preproc_threads,
        dali_cpu=dali_cpu, idx_filenames=eval_idx_filenames)
```

File: docs/examples/tensorflow/demo/nvutils/runner.py (hunk 3 above)
Comment (Kh4L) [reply docs/examples/tensorflow/demo/nvutils/runner.py]: Done

File: docs/examples/tensorflow/demo/nvutils/runner.py (hunk 4 above)
Comment (Kh4L) [reply docs/examples/tensorflow/demo/nvutils/runner.py]: Done

**Comment by JanuszL** [comment]
Builds 38608

File: qa/L1_tensorflow-dali_test/test.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+cd docs/examples/tensorflow/demo
+
+mkdir -p idx-files/
+
+NUM_GPUS=$(nvidia-smi -L | wc -l)
+
+test_body() {
+    # test code
+
+    OPENMPI_VERSION=3.0.0
```
[the remark this answers is not in the document]
Comment (Kh4L) [reply qa/L1_tensorflow-dali_test/test.sh]: Done
